### PR TITLE
INTYGFV-16529: Added storage of ExternalReference. Added errorMessage to GlobalExceptionHandlerController for CertificateActionForbidden.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Comment by ChrilleSigerhed** [comment]
Example error response with two errors:

```xml
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <ns4:CreateDraftCertificateResponse>
      <ns4:result>
        <ns5:resultCode>ERROR</ns5:resultCode>
        <ns5:resultText>403 : "Du saknar behörighet för den begärda åtgärden. Det krävs särskilda rättigheter eller en specifik befattning
          för att hantera patienter med skyddade personuppgifter., Du saknar behörighet för den begärda åtgärden. Det krävs särskilda
          rättigheter eller en specifik befattning för att hantera avlidna patienter."
        </ns5:resultText>
        <ns5:errorId>VALIDATION_ERROR</ns5:errorId>
      </ns4:result>
    </ns4:CreateDraftCertificateResponse>
  </soap:Body>
</soap:Envelope>
```

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java [hunk 1]
```diff
@@ -37,6 +38,7 @@ public class Certificate {
   private Xml xml;
   private Sent sent;
   private Revoked revoked;
+  private ExternalReference externalReference;
```
Comment (mww-sopra-steria) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java:41]: ExternalReference is both in Certificate and CertificateMetadata. I would like to see it as part of Certificate only.

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java [hunk 2]
```diff
@@ -68,6 +70,7 @@ public void updateMetadata(ActionEvaluation actionEvaluation) {
         .careUnit(actionEvaluation.careUnit())
         .careProvider(actionEvaluation.careProvider())
         .issuingUnit(actionEvaluation.subUnit())
+        .externalReference(actionEvaluation.externalReference())
```
Comment (mww-sopra-steria) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/certificate/model/Certificate.java:73]: Not sure that externalReference should be updated if already provided. Do you know? Best to double check with Webcert and current implementation.

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/common/exception/CertificateActionForbidden.java [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+import java.util.Collections;
+import java.util.List;
+import lombok.Getter;
+
+@Getter
 public class CertificateActionForbidden extends RuntimeException {
 
+  private final List<String> reason;
+
   public CertificateActionForbidden(String message) {
```
Comment (mww-sopra-steria) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/common/exception/CertificateActionForbidden.java:12]: Suggest we remove the default-constructor and force all places to provide reason.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Nice work on this PR! Agree that it would be good in Webcert to remove the status-code. Also I noticed that we don't log the exception, that needs to be done! :-)

Relating to this PR there are a few minor things to look at. But also a couple of things that I would appreciate if we could discuss.

**Review by mww-sopra-steria — APPROVED** [review]
Looks great! Well done!

**Pull request merged**